core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -49,16 +49,17 @@ class KryoSerializer(conf: SparkConf)
   with Logging
   with Serializable {
 
-  private val bufferSizeMb = conf.getDouble("spark.kryoserializer.buffer.mb", 0.064)
-  if (bufferSizeMb >= 2048) {
-    throw new IllegalArgumentException("spark.kryoserializer.buffer.mb must be less than " +
-      s"2048 mb, got: + $bufferSizeMb mb.")
+  private val bufferSizeKb = conf.getSizeAsKb("spark.kryoserializer.buffer", "64k")
```

Comment:
All - I don't know what the right solution here is. The old value simply can't work using the new framework. Fractional values are no longer supported and near as I can tell this is is the only instance of such usage. The only way to truly maintain backwards compatibility (short of throwing an exception) is to leave this as conf.getDouble but then this is an exception to the rule for how we handle size variables. 

This needs to be ```getSizeAsKb()``` and the value must be specified in the right format, otherwise an exception is thrown.